CONTRIBUTING.md

```diff
@@ -0,0 +1,139 @@
+# Contributing to Eclair
+
+Eclair welcomes contributions in the form of peer review, testing and patches.
+This document explains the practical process and guidelines for contributing.
+
+While developing a Lightning implementation is an exciting project that spans many domains
+(cryptography, peer-to-peer networking, databases, etc), contributors must keep in mind that this
+represents real money and introducing bugs or security vulnerabilities can have far more dire
+consequences than in typical projects. In the world of cryptocurrencies, even the smallest bug in
+the wrong area can cost users a significant amount of money.
+
+If you're looking for somewhere to start contributing, check out the [good first issue](https://github.com/acinq/eclair/issues?q=is%3Aopen+is%3Aissue+label%3A"good+first+issue") list.
+
+Another way to start contributing is by adding tests or improving them.
+This will help you understand the different parts of the codebase and how they work together.
+
+## Communicating
+
+We recommend using our Gitter [developers channel](https://gitter.im/ACINQ/developers).
+Introducing yourself and explaining what you'd like to work on is always a good idea: you will get
+some pointers and feedback from experienced contributors. It will also ensure that you're not
+duplicating work that someone else is doing.
+
+We use Github issues only for, well, issues (mostly bugs that need to be investigated).
+You can also use Github issues for [feature requests](https://github.com/acinq/eclair/issues?q=is%3Aissue+label%3A"feature+request").
+
+## Recommended Reading
+
+- [Bitcoin Whitepaper](https://bitcoin.org/bitcoin.pdf)
+- [Lightning Network Whitepaper](https://lightning.network/lightning-network-paper.pdf)
+- [Deployable Lightning](https://github.com/ElementsProject/lightning/raw/master/doc/deployable-lightning.pdf)
+- [Understanding the Lightning Network](https://bitcoinmagazine.com/articles/understanding-the-lightning-network-part-building-a-bidirectional-payment-channel-1464710791)
+- [Lightning Network Specification](https://github.com/lightningnetwork/lightning-rfc)
+
```

Comment:
When reading about the protocol i personally found very valuable this small series of BOLT protocol explanations https://medium.com/@rusty_lightning/the-bitcoin-lightning-spec-part-1-8-a7720fb1b4da might be worth adding them

Reply:
I wasn't lucky enough to stumble on that one when I started :D
Let's add it to help newcomers!